### PR TITLE
Refs #33061 -- Removed unnecessary BaseMemcachedCache.decr().

### PR DESCRIPTION
**Pull request by felixxm**
We can reuse `BaseMemcachedCache.incr()` logic.

### CODE REVIEW

File: django/core/cache/backends/memcached.py [hunk 1]
```diff
@@ -105,14 +105,14 @@ def close(self, **kwargs):
...
         try:
-            val = self._cache.incr(key, delta)
-
+            # memcached doesn't support a negative delta
+            if delta < 0:
+                val = self._cache.decr(key, -delta)
+            else:
+                val = self._cache.incr(key, delta)
         # Normalize an exception raised by the underlying client library to
         # ValueError in the event of a nonexistent key when calling incr().
```
Comment (ngnpope) [inline django/core/cache/backends/memcached.py:117]: This comment should be amended to reflect that it is a call to `incr()` or `decr()` of the underlying client now they're in the same block.

File: django/core/cache/backends/memcached.py [hunk 2]
```diff
@@ -105,14 +105,14 @@ def close(self, **kwargs):
...
-        # memcached doesn't support a negative delta
-        if delta < 0:
-            return self.decr(key, -delta, version=version)
         key = self.make_key(key, version=version)
         self.validate_key(key)
         try:
-            val = self._cache.incr(key, delta)
-
+            # memcached doesn't support a negative delta
+            if delta < 0:
```
Comment (ngnpope) [inline django/core/cache/backends/memcached.py:112]: There is a minor behaviour change here. Previously calling `decr()` with `delta=0` would call `self._cache.decr()`, but now it'll call `self._cache.incr()` instead.

In theory this shouldn't be a problem, but am highlighting it.

File: django/core/cache/backends/memcached.py (hunk 1 above)
Comment (felixxm) [reply django/core/cache/backends/memcached.py]: Updated.

File: django/core/cache/backends/memcached.py (hunk 2 above)
Comment (felixxm) [reply django/core/cache/backends/memcached.py:112]: I don't think that's an issue.

**Review by ngnpope — APPROVED** [review]
Nice little clean up. Thanks!

File: django/core/cache/backends/memcached.py [hunk 3]
```diff
@@ -105,33 +105,17 @@ def close(self, **kwargs):
...
-    def decr(self, key, delta=1, version=None):
-        # memcached doesn't support a negative delta
-        if delta < 0:
-            return self.incr(key, -delta, version=version)
-        key = self.make_key(key, version=version)
-        self.validate_key(key)
-        try:
-            val = self._cache.decr(key, delta)
-
+            # memcached doesn't support a negative delta
```
Comment (cjerdonek) [inline django/core/cache/backends/memcached.py:111]: Might as well switch the comment to PEP 8:

```python
# Memcached doesn't support negative delta.
```

**Review by cjerdonek — APPROVED** [review]
👍

**Comment by felixxm** [comment]
Thanks both for reviews :+1:

**Pull request merged**